pkg/resource/deploy/deployment_executor.go

```diff
@@ -32,6 +33,8 @@ import (
 	"github.com/pulumi/pulumi/sdk/v3/go/common/util/result"
 )
 
+var UseParallelStepGen = false
```

Comment:
Could we pass this through engine options instead?  Or maybe just use the env variable since that's already global state, so we don't need to duplicate it here?

Reply:
+1 to threading this through options; -1 to using the envvar. I'd love to see the former to make this a bit more hermetic / testable.

Reply:
The purpose of this is to be able to globally toggle the behavior for testing, I agree that passing it through the engine options would be cleaner for the non-testing uses